### PR TITLE
YAML: Support macvlan configuration using yml

### PR DESCRIPTION
**Pull request by ntsbtz**
Also cmocka test to add multiple macvlan
Invalid Bond mode fix

### CODE REVIEW

**Comment by ssahani** [comment]
this needs to be rebased

File: src/yaml/yaml-netdev-parser.c [hunk 1]
```diff
@@ -815,6 +875,17 @@ int yaml_parse_netdev_config(YAMLManager *m, YAMLNetDevKind kind, yaml_document_
 int yaml_register_netdev(YAMLManager *m) {
         assert(m);
 
+        m->macvlan = g_hash_table_new(g_str_hash, g_str_equal);
+        if (!m->macvlan)
```
Comment (ssahani) [inline src/yaml/yaml-netdev-parser.c:879]: `m->macvlan ` is not freed ?

File: src/yaml/yaml-parser.c [hunk 2]
```diff
@@ -459,6 +459,26 @@ int parse_yaml_bond_mode(const char *key,
         return 0;
 }
 
+int parse_yaml_macvlan_mode(const char *key,
+                            const char *value,
+                            void *data,
+                            void *userdata,
+                            yaml_document_t *doc,
+                         yaml_node_t *node) {
+        MACVLan *m;
```
Comment (ssahani) [inline src/yaml/yaml-parser.c:468]: allignment

**Review by ssahani — CHANGES_REQUESTED** [review]
Please also add to the documentation

File: src/manager/netdev.h [hunk 3]
```diff
@@ -262,6 +262,7 @@ typedef struct Veth {
 } Veth;
 
 typedef struct MACVLan {
+         char *master;
          MACVLanMode mode;
```
Comment (ssahani) [inline src/manager/netdev.h:266]: also you need to free the `master`

Comment (ntsbtz) [reply src/manager/netdev.h:266]: oh my bad missed it. Resolved !!!

**Pull request merged**